### PR TITLE
Add support for Redis ZADD command for RedisIO Write

### PR DESCRIPTION
**Pull request by anish749**
We are trying out RedisIO with ZAdd command, and it was missing in here.

### CODE REVIEW

**Review by stormy-ua — APPROVED** [review]
👍

File: scio-redis/src/main/scala/com/spotify/scio/redis/types/RedisMutation.scala [hunk 1]
```diff
@@ -54,3 +54,13 @@ final case class RPush[T](key: T, value: Seq[T], ttl: Option[Duration] = None)(i
...
 ) extends RedisMutation
+final case class ZAdd[T](key: T, scoreMembers: Map[T, Double], ttl: Option[Duration] = None)(
+  implicit val rt: RedisType[T]
+) extends RedisMutation
+object ZAdd {
+  def apply[T: RedisType](key: T, score: Double, member: T): ZAdd[T] =
+    ZAdd(key, Map(member -> score))
+
+  def apply[T: RedisType](key: T, score: Double, member: T, ttl: Option[Duration]): ZAdd[T] =
+    ZAdd(key, Map(member -> score))
```
Comment (stormy-ua) [inline scio-redis/src/main/scala/com/spotify/scio/redis/types/RedisMutation.scala:65]: Should `ttl` be passed down to `ZAdd` here?

Comment (stormy-ua) [inline scio-redis/src/main/scala/com/spotify/scio/redis/types/RedisMutation.scala:65]:
```suggestion
    ZAdd(key, Map(member -> score), ttl)
```

Comment (anish749) [reply scio-redis/src/main/scala/com/spotify/scio/redis/types/RedisMutation.scala]: Good catch :) it should be passed. Thanks.

File: scio-redis/src/main/scala/com/spotify/scio/redis/types/RedisMutation.scala [hunk 2]
```diff
@@ -54,3 +54,13 @@ final case class RPush[T](key: T, value: Seq[T], ttl: Option[Duration] = None)(i
 final case class PFAdd[T](key: T, value: Seq[T], ttl: Option[Duration] = None)(implicit
   val rt: RedisType[T]
 ) extends RedisMutation
+final case class ZAdd[T](key: T, scoreMembers: Map[T, Double], ttl: Option[Duration] = None)(
+  implicit val rt: RedisType[T]
+) extends RedisMutation
+object ZAdd {
+  def apply[T: RedisType](key: T, score: Double, member: T): ZAdd[T] =
```
Comment (regadas) [inline scio-redis/src/main/scala/com/spotify/scio/redis/types/RedisMutation.scala:61]:
```suggestion
  final def apply[T: RedisType](key: T, score: Double, member: T): ZAdd[T] =
```

File: scio-redis/src/main/scala/com/spotify/scio/redis/types/RedisMutation.scala [hunk 3]
```diff
@@ -54,3 +54,13 @@ final case class RPush[T](key: T, value: Seq[T], ttl: Option[Duration] = None)(i
...
+    ZAdd(key, Map(member -> score))
+
+  def apply[T: RedisType](key: T, score: Double, member: T, ttl: Option[Duration]): ZAdd[T] =
```
Comment (regadas) [inline scio-redis/src/main/scala/com/spotify/scio/redis/types/RedisMutation.scala:64]:
```suggestion
  final def apply[T: RedisType](key: T, score: Double, member: T, ttl: Option[Duration]): ZAdd[T] =
```

**Review by regadas — APPROVED** [review]

**Pull request merged**